packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -225,6 +225,45 @@ export interface CanaryProps {
    */
   readonly test: Test;
 
+  /**
+   * Key-value pairs that the Synthetics caches and makes available for your canary
+   * scripts. Use environment variables to apply configuration changes, such
+   * as test and production environment configurations, without changing your
+   * Canary script source code.
+   *
+   * @default - No environment variables.
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * The function execution time (in seconds) after which Lambda terminates
+   * the function. Because the execution time affects cost, set this value
+   * based on the function's expected execution time.
+   *
+   * @default cdk.Duration.seconds(900)
+   */
+  readonly timeout?: cdk.Duration;
+
+  /**
+   * The amount of memory, in MB, that is allocated to your Lambda function.
+   * Lambda uses this value to proportionally allocate the amount of CPU
+   * power. For more information, see Resource Model in the AWS Lambda
+   * Developer Guide.
+   *
+   * @default 128
+   */
+  readonly memorySize?: number;
+
+  /**
+   * Specifies whether this canary is to use active AWS X-Ray tracing when it runs.
+   * Active tracing enables this canary run to be displayed in the ServiceLens and
+   * X-Ray service maps even if the canary does not hit an endpoint that has
+   * X-ray tracing enabled.
+   *
+   * @default false
+   */
+  readonly tracing?: boolean;
```

Comment:
You need to give the canary permissions described here: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_tracing.html
Also would be good to mention in your documentation that enabling this option `increases canary run time by 2.5% to 7%.`

Reply:
Added

Reply:
please note that if a role is provided, it must also have permissions documented here: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_tracing.html